cores/esp8266/flash_hal.h

```diff
@@ -33,21 +33,32 @@ extern "C" {
 #include <FlashMap.h>
 
 extern uint32_t spi_flash_get_id (void); // <user_interface.h>
-extern void flashinit(void);
+extern const char *flashinit(void);
 extern uint32_t __flashindex;
 extern const flash_map_s __flashdesc[];
 
+#ifndef QUOTE
+#define QUOTE(a) __STRINGIFY(a)
+#endif
+
+#ifdef DEBUG_ESP_CORE
+#define DEBUG_FLASH_MAP_NOT_FOUND " flashinit: configuration not found"
+#else
+#define DEBUG_FLASH_MAP_NOT_FOUND
+#endif
```

Comment:
we always print exc text though, why hide?
could also just have `static const char DEBUG_FLASH_MAP_NOT_FOUND[] PROGMEM = ...` right here at the top?

Reply:
Was concerned about growing space used up in non-debug builds.

I was thinking I wanted the line # reported to be in the function body; however, I later realized it is the macro line number that is being reported. Also, if I place it in the scope of the function I avoid the potential problem of a hidden variable name conflict in the module.

I have no strong preferences.

Reply:
Maybe file and line are unnecessary in this case, and only the string is relevant and useful even in non-debug mode.

Reply:
That was my first attempt then I realized the user could provide a broken override `flashinit`. 
Providing the file and line gives a strong indication of where the fault lives. And that the override won during the linking process.